airflow/contrib/operators/neo4j_operator.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Neo4JOperator to interact and perform action on Neo4J graph database.
+This operator is designed to use Neo4J Hook and the
+Python driver: https://neo4j.com/docs/api/python-driver/current/
+"""
+
+import csv
+from os.path import isfile
+
+from neo4j import BoltStatementResult
+
+from airflow.contrib.hooks.neo4j_hook import Neo4JHook
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class Neo4JOperator(BaseOperator):
+    """
+    This operator provides Airflow DAGs the ability to execute a cypher query
+    and save the results of the query to a CSV file.
+    :param cypher_query: required cypher query to be executed on the Neo4J database
+    :type cypher_query: str
+    :param output_filename: required filename to produce with output from the query
+    :type output_filename: str
+    :param n4j_conn_id: reference to a pre-defined Neo4J Connection
+    :type n4j_conn_id: str
+    :param soft_fail: True/False flag to indicate if it should fail the task if no results
+    :type soft_fail: bool
+    """
+    _cypher_query = None
+    _output_filename = None
+    _n4j_conn_id = None
+    _soft_fail = None
+
+    template_fields = ['cypher_query', 'output_filename', 'n4j_conn_id', 'soft_fail']
```

Comment:
This won't work. Template fields should be names of instance attributes that should be rendered. So in this case:
```python
template_fields = ['_cypher_query', '_output_filename', _'n4j_conn_id', '_soft_fail']
```
That's why I suggested removing the underscore.

Reply:
Thanks, I understand now. I've removed the underscore prefix from the class variables now.
Running the tests locally, if it all passes I'll push the change up shortly.
Thanks for clarifying.